### DISCUSSION TITLE
links from CSV to code

### DISCUSSION THREAD

**Discussion by mjt320** [OSIPI/DCE-DSC-MRI_CodeCollection#114] [Ideas] [answered]
Hi all. Steven commented on Slack that it was not obvious how to find source code.
User experience is obviously something we could think about for the new roadmap.
In the meantime, I thought we could add hyperlinks from the CSV file to the source code folders.
It could serve as a "quick fix" before the manuscript goes in for review.
What do you think?

**Comment by LucyKershaw** [comment]
That seems sensible to me as an interim fix.  Making code contribution easier is certainly high on the priority list but that's useless if you can't easily find what's been submitted.  

How difficult is that to do? Is it very manual?

**Reply by mjt320** [reply]
I was thinking of just pasting links into the table for now

**Comment by LucyKershaw** [comment]
Do you have time to do it?  Hopefully we can come up with something more automatic for the future

**Reply by mjt320** [reply]
Happy to give it a go. It _might_ turn out to be easy and quick...

**Reply by LucyKershaw** [reply]
Ok, let me know if it isn't and you need an extra pair of hands

**Comment by mjt320** [comment]
A major issue is that github won't render hyperlinks within a csv file!

**Comment by LucyKershaw** [comment]
Can you at least right click and follow the link text?

**Reply by mjt320** [reply]
You can highlight then right click -> "go to..."

**Comment by LucyKershaw** [comment]
Adam sent me an example of a github page where they had a table rendered
with links but it was written in markdown.  Let's discuss at the TF meeting
but I think it might be the way to go.  I could learn how to do it I thnk

**Comment by mjt320** [comment]
I've persuaded github actions to convert the CSV file to markdown.
See:
https://github.com/OSIPI/DCE-DSC-MRI_CodeCollection/blob/add_links_to_csv/doc/code_contributions_record.md
(only added a link to the first entry)
Would this be a good solution?

**Reply by stadmill** [reply]
Just tested it and it works fine. Do you need any help converting the other entries?

**Reply by LucyKershaw** [reply]
This is brilliant, thanks Michael - is it difficult?

**Reply by mjt320** [reply]
Should be easy now that it's set up.
I've added a github actions step, which runs a python script:
https://github.com/OSIPI/DCE-DSC-MRI_CodeCollection/blob/add_links_to_csv/doc/convert_to_md.py
This converts the CSV to an MD.

All we need to do is add links to the CSV, e.g.
\[piecewise linear quadratic function](../src/original/LEK_UoEdinburgh_UK/PharmacokineticModelling/BAT_estimation.py)
I suggest using relative paths, as above.
I have no objection to others helping to add the links (use the add_links branch) :)

**Comment by stadmill** [comment]
Just found this on GitHub Docs on creating permanent links to files in repositories - [see here](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files )

**Reply by mjt320** [reply]
Thanks, good to know.
I imagine for this application we don't want permanent links, in case the source code is updated

**Comment by mjt320** [comment]
Hello.
I have added links to the CSV file and made a pull request. The markdown conversion didn't run...hopefully it will run when the pull request is merged.

**Reply by stadmill** [reply]
Hi Michael, I just merged it. I can see the link in the md file now, the link works perfectly and takes me to the source code.

**Comment by mjt320** [comment]
Hi @sunyu0410
I've added a workflow to convert our CSV file to MD format (so that links are displayed).
It was working before (and works locally) but doesn't seem to be working now.
It's probably down to my ignorance of git/actions. Are you able to help?
Thanks, Michael

**Reply by mjt320** [reply]
Actually, I've managed to get it working now following a bit of googling.
The MD file now updates and has links on every line

**Comment by mjt320** [comment, marked as answer]
Done
